pulsar-websocket/src/main/java/com/yahoo/pulsar/websocket/AbstractWebSocketHandler.java

```diff
@@ -73,11 +74,16 @@ public void onWebSocketConnect(Session session) {
             }
         }
 
-        if (service.isAuthorizationEnabled() && !isAuthorized(authRole)) {
-            log.warn("[{}] WebSocket Client [{}] is not authorized on topic {}", session.getRemoteAddress(), authRole,
-                    topic);
-            close(WebSocketError.NotAuthorizedError);
-            return;
+        if (service.isAuthorizationEnabled()) {
+            final String role = authRole;
+            isAuthorized(authRole).thenApply(isAuthorized -> {
```

Comment:
Yes.. ```WebSocketHandler.onWebSocketConnect()``` gets triggered after ```HttpConnection``` is established and as soon as ```HttpConnection``` is established : client also receives ```onConnect``` event.
So,  ```HttpConnection.onCompleted()``` triggers ```WebSocketHandler.onWebSocketConnect()```.
So, by the time  ```WebSocketHandler``` receives ```onWebSocketConnect()``` notification, client has already received ```onConnect()``` event. 

- However, after fixing [websocket-test-case](https://github.com/yahoo/pulsar/pull/99), I have verified that ```onWebSocketText(..)``` gets triggered at server only when server releases thread from ```onWebSocketConnect()``` So, reverting back to sync-zk behavior will make sure that producer will start receiving new messages on ```onWebSocketText()``` only after completion of authorization at ```onWebSocketConnect()```.